### PR TITLE
fix bug when can not get pod by ip in k8s registry

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
fixes: #9515

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller.go [hunk 1]
```diff
@@ -485,30 +485,29 @@ func (c *Controller) InstancesByPort(hostname model.Hostname, reqSvcPort int,
...
-	// 1. find proxy service by label selector, if not any, there may exist headless service
-	// failover to 2
-	svcLister := listerv1.NewServiceLister(c.services.informer.GetIndexer())
-	if services, err := svcLister.GetPodServices(pod); err != nil && len(services) > 0 {
-		for _, svc := range services {
-			item, exists, err := c.endpoints.informer.GetStore().GetByKey(KeyFunc(svc.Namespace, svc.Name))
-			if err != nil || !exists {
-				log.Warnf("unable to get endpoint %s/%s for svc: %v", svc.Namespace, svc.Name, err)
-				continue
+				ep := *item.(*v1.Endpoints)
```
Comment (costinm) [inline pilot/pkg/serviceregistry/kube/controller.go]: AFAIK the code that is calling 'getProxyServiceInstances' doesn't actually care or use the endpoints - the method is used to compute inbound listeners/clusters. The actual endpoints are used in EDS - but that's not using this method. 

Maybe changing the signature or adding a separate method, or just deprecating / removing the use of this method and writing a new one in PushContext or ADS, using the 'workload' callback and computing only what is needed in code generation for inbound. 

But separate PR.

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller.go]: That is a big optimization, will do it later.

**Comment by hzxuzhonghu** [comment]
Since will cherry-pick back from release-1.1, close this one.

**Pull request closed without merging**